### ISSUE TITLE
Bug validating scientific notation

### ISSUE THREAD

**Issue by rigal-m** [23andMe/Yamale#34]
Hello,
I am encountering an issue when I have a float written in the scientific notation. For example, under python3:

```yaml
# [data.yaml]
key1: 1.0e-6
key2: 1e-6
```

```yaml
# [schema.yaml]
key1: num()
key2: num()
```

When I try to validate with the command `yamale data.yaml`, I get the error:
```
ValueError: 
Error validating data /local/home/mrigal/Work/parsing_yaml/issues/issue1/data.yaml with schema schema.yaml
	key2: '1e-6' is not a num.
```

For some reason it seems that PyYAML converts 1e-6 to a string (but not 1.0e-6). If I modify the source file yamale/readers/yaml_reader.py to load the module ruamel.yaml instead of yaml, it works fine:

```python
from __future__ import absolute_import
from ruamel.yaml import YAML

yaml=YAML(typ="safe")

def parse_file(file_name):
    with open(file_name) as f:
        return list(yaml.load_all(f))
```

**Comment by blopker** [comment]
Thank you for the report.

It looks like the main issue is that PyYAML does not support YAML 1.2, where ruamel.yaml does. Work on PyYAML has been very slow recently and I'm not sure if/when 1.2 will be supported. So maybe Yamale could conditionally support both parsers and the user can select which one they want.

Would you be interested in creating a PR for that?

**Comment by rigal-m** [comment]
Yes, I will take care of it asap.

**Comment by blopker** [comment]
Fixed by #36
